crypto/encode_decode/encoder_lib.c

```diff
@@ -49,6 +49,49 @@ int OSSL_ENCODER_to_fp(OSSL_ENCODER_CTX *ctx, FILE *fp)
 }
 #endif
 
+int OSSL_ENCODER_to_data(OSSL_ENCODER_CTX *ctx, unsigned char **pdata,
+                         size_t *pdata_len)
+{
+    BIO *out = BIO_new(BIO_s_mem());
+    BUF_MEM *buf = NULL;
+    int ret = 0;
+
+    if (pdata_len == NULL) {
+        ERR_raise(ERR_LIB_OSSL_DECODER, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    if (OSSL_ENCODER_to_bio(ctx, out)
+        && BIO_get_mem_ptr(out, &buf) > 0) {
+        ret = 1; /* Hope for the best. A too small buffer will clear this */
+
+        if (pdata != NULL && *pdata != NULL) {
+            if (*pdata_len < buf->length)
+                ret = 0;
```

Comment:
It could be useful to also do *pdata_len = buf->length in this case however it would complicate the API contract. So this is just for your consideration.

Reply:
No, that's actually a good idea, and would help the user to allocate an appropriate buffer even in that case.

Reply:
Hmmm, on second thought, you're right, that might actually be more confusing than helpful in this case.  So I'm backing away from that idea, but will add comment.